src/osdc/Objecter.h

```diff
@@ -2123,8 +2123,7 @@ class Objecter : public md_config_obs_t, public Dispatcher {
   // here or you will have great woe and misery.
 
   template<typename Callback, typename...Args>
-  auto with_osdmap(Callback&& cb, Args&&... args) const ->
-    decltype(cb(*osdmap, std::forward<Args>(args)...)) {
+  decltype(auto) with_osdmap(Callback&& cb, Args&&... args) {
```

Comment:
```suggestion
  auto with_osdmap(Callback&& cb, Args&&... args) {
```

Reply:
https://en.cppreference.com/w/cpp/language/auto tells perfect forwarding of a function call must use decltype(auto) in case the function it calls returns by reference